src/operations/identify.rs

```diff
@@ -1,18 +1,41 @@
-use crate::{error::MagickError, image::Image};
+use crate::{error::MagickError, image::Image, wm_try};
+use std::ffi::{OsStr, OsString};
+use std::io::{self, Write};
 
+// https://imagemagick.org/script/command-line-options.php#identify
 pub fn identify(image: &mut Image) -> Result<(), MagickError> {
-    println!("{}", identify_impl(image));
+    // use a writer to write out the raw bytes to stdout
+    // so that we can support non-UTF8 filenames as is possible on unix.
+    let info = identify_impl(image);
+    wm_try!(io::stdout().write_all(info.as_encoded_bytes()));
+    wm_try!(io::stdout().write_all(b"\n"));
     Ok(())
 }
 
-fn identify_impl(image: &Image) -> String {
-    format!("{}x{}", image.pixels.width(), image.pixels.height())
+fn identify_impl(image: &Image) -> OsString {
+    let filename = Some(image.properties.filename.clone());
```

Comment:
Probably not the most memory-efficient implementation route I took here, but is this a "hot path"? Not necessarily I would say, maybe we can live with a couple of clones for the moment?